cmd/root.go

```diff
@@ -272,3 +270,25 @@ func setupImagePullSecretsProvider() secrets.ImagePullSecretsProvider {
 
 	return secrets.NewKubernetesImagePullSecretsProvider(clientset)
 }
+
+// setupRegistry configures a target registry client connection
+func setupTargetRegistryClient() (registry.Client, error) {
+	targetRegistry, err := types.ParseTargetRegistry(cfg.Target.Type)
+	if err != nil {
+		log.Err(err)
+	}
+
+	switch targetRegistry {
+	case types.TargetRegistryGcp:
+		return registry.NewGARClient(cfg.Target.GCP.Location, cfg.Target.GCP.ProjectID, cfg.Target.GCP.RepositoryID)
+	case types.TargetRegistryAws:
+		fallthrough
+	default:
+		registry, err := registry.NewECRClient(cfg.Target.AWS.Region, cfg.Target.AWS.EcrDomain(), cfg.Target.AWS.AccountID, cfg.Target.AWS.Role, cfg.Target.AWS.ECROptions.AccessPolicy, cfg.Target.AWS.ECROptions.LifecyclePolicy)
+		if err != nil {
+			return registry, err
+		}
+		registry.SetRepositoryTags(cfg.Target.AWS.ECROptions.Tags)
+		return registry, err
+	}
```

Comment:
Imo, `case types.TargetRegistryAWS: fallthrough` is not required here. If you leave it out the default will be ECR.

I think I'd prefer the code block to move to `case types.TargetRegistryAWS` and keep `default` to raise an error if the type of registry is unsupported, e.g.:

```suggestion
	case types.TargetRegistryAws:
		registry, err := registry.NewECRClient(cfg.Target.AWS.Region, cfg.Target.AWS.EcrDomain(), cfg.Target.AWS.AccountID, cfg.Target.AWS.Role, cfg.Target.AWS.ECROptions.AccessPolicy, cfg.Target.AWS.ECROptions.LifecyclePolicy)
		if err != nil {
			return nil, err
		}
		registry.SetRepositoryTags(cfg.Target.AWS.ECROptions.Tags)
		return registry, err
	default:
	         return nil, errors.New("unsupported target registry")
	}
```

I also would not return the instantiated registry if an error occurs.

Happy for you to adjust this or disagree :)

Reply:
Regarding the `fallthrough` and error case returning an instantiated registry, I will make the proposed changes.

Regarding raising an error for unsupported type. My initial decision here was to keep backwards compatibility with folks that are primarily using `k8s-image-swapper` with ECR. At this time they do not need to have a `target.type` defined in their config.

If we decide to adjust the default (and enforce a valid type) we'll likely want to make that clear as part of upgrade instructions and bump the major version of the project to `2.0.0`?

Reply:
Fair enough. Let's bump it to `2.0.0` if it is incompatible and provide documentation on what needs to be updated.

As alternative we can set the default for `target.type` to `aws`:

```golang
viper.SetDefault("target.type", "aws")
```

This is being set here: https://github.com/estahn/k8s-image-swapper/blob/66692fd17fe1d7573a61556c8e67fa62b45b8f6f/cmd/root.go#L208-L209

Reply:
I updated the code to throw an error if the `target.type` is not set.

If there are other changes you know of to prep for a 2.0.0 release that I need to do, please let me know.

Reply:
I updated the code to default `target.type` to `aws` via viper.  We should be good to continue with a `1.5.0` release.